concent_api/gatekeeper/views.py

```diff
@@ -154,6 +158,30 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
     if len(transfer_token_paths_to_files) != len(set(transfer_token_paths_to_files)):
         return gatekeeper_access_denied_response('File paths in the token must be unique', path_to_file, loaded_golem_message.subtask_id, client_public_key)
 
+    if any((file["checksum"] is None or len(file["checksum"]) == 0 or file["checksum"].isspace()) for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The variable is empty', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+    if any(not isinstance(file["checksum"], str) for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The variable must be string', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+    file_checksums = [tuple(file["checksum"].split(":")) for file in loaded_golem_message.files]
+    if any(file_checksum[0] != hashing_algorytm.value for file_checksum in file_checksums for hashing_algorytm in HashingAlgorytm):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The hashing algorytm is not supported', path_to_file, loaded_golem_message.subtask_id, client_public_key)
```

Comment:
This could be simpler:

``` python
if any(checksum_type not in HashingAlgorithm.__members__ for checksum_type, checksum in checksums):
```

or

``` python
if not set(checksum_type for checksum_type, checksum in checksums).issubset(set(HashingAlgorithm.__members__)):
```

Reply:
Object `HashingAlgorithm.__members__` return order dictionary `OrderedDict([('SHA1', <HashingAlgorithm.SHA1: 'sha1'>)])` and not work with yours example I would stay with an earlier solution. What you think ?